index.js

```diff
@@ -157,6 +157,18 @@ class Executor {
         throw new Error('Not implemented');
     }
 
+    /**
+     * Clean up any processing tasks
+     * @method cleanUp
+     */
+    async cleanUp() {
+        await this._cleanUp();
+    }
+
+    async _cleanUp() {
+        // no-op in case no implemented in extenders
```

Comment:
is it proper to enforce mandatory clean up?

Reply:
Yes, tried to keep changes to minimal